llvm/lib/Transforms/Scalar/LoopInterchange.cpp

```diff
@@ -156,7 +158,13 @@ static bool populateDependencyMatrix(CharMatrix &DepMatrix, unsigned Level,
           Dep.push_back('I');
         }
 
-        DepMatrix.push_back(Dep);
+        // Make sure we only add unique entries to the dependency matrix.
+        std::string DepStr = std::string(Dep.begin(), Dep.end());
```

Comment:
I think you should be able to create this as a StringRef. No need to allocate std::string here.